benchmarking/blackbox_repository/blackbox_surrogate.py

```diff
@@ -31,6 +31,7 @@ def __init__(
             fidelity_space: Optional[Dict] = None,
             fidelity_values: Optional[np.array] = None,
             surrogate=KNeighborsRegressor(n_neighbors=1),
+            max_fit_samples: Optional[int] = None,
```

Comment:
Can you document this?

BTW: I'd not downsample the BB data, these nonparametric methods can handle it, esp. KNN.

If this is about reducing the size of the dill file, let us do this in another way. I have a PR open, and maybe you have another idea.

But I view this BB data as really valuable, and the goal is interpolation, so you really do not want to subsample. One can subsample for things like classification, but interpolating a complex function like this, you really want to use all the data you have.

Reply:
Added a comment. This is not about reducing the file of the dill file, we are only saving the surrogate name and its arguments, it is to reduce the potential very long runtime that happens if you have a large number of evaluations.